lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -411,6 +419,7 @@ impl Client {
                         if res.err_code == TdpErrCode::Nil {
                             return cli.tdp_sd_overwrite(rdp_req, res.fso);
                         } else {
+                            // The match statement on res.err_code above ensures that this means res.err_code == TdpErrCode::DNE
```

Comment:
Hmmm, these seems fragile. As soon as you change the branches of the match this breaks. Should we just be more explicit and compare to DNE?

Reply:
Yeah, I believe the reason I opted for this initially is that I didn't want the compiler to force me to add this line:

https://github.com/gravitational/teleport/blob/aa1af3a8e8c4965e84ec09e1b8b4d9bd6a5d6b88/lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs#L494

which is needed if those statements are `else if`'s rather than `else`'s.

I've played around with the logic (like wrapping everything in a `match res.err_code`), and unfortunately haven't come up with a configuration that's logically sound and where the compiler is smart enough to understand it, except for this.